File: Examples.VB/Features/FindAndReplace/FindWithAfter.vb
Namespace Features.FindAndReplace
    Public Class FindWithAfter
        Inherits ExampleBase

        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)

            ' Prepare data
            Const CorrectWord = "Macro"
            worksheet.Range("A1:D5").Value = CorrectWord

            Const MisspelledWord = "marco"
            worksheet.Range("A2,C3,D1").Value = MisspelledWord

            ' Find all misspelled words and mark them with red background
            Dim searchRange As IRange = worksheet.Range("A1:D5")
            Dim misspelledCell As IRange = Nothing
            Do
                misspelledCell = searchRange.Find(MisspelledWord, misspelledCell)
                If misspelledCell Is Nothing Then
                    Exit Do
                End If
                misspelledCell.Interior.Color = System.Drawing.Color.Red
            Loop
        End Sub

        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace

File: Examples/Features/FindAndReplace/FindWithAfter.cs
namespace GrapeCity.Documents.Excel.Examples.Features.FindAndReplace
{
    public class FindWithAfter : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];

            // Prepare data
            const string CorrectWord = "Macro";
            worksheet.Range["A1:D5"].Value = CorrectWord;

            const string MisspelledWord = "marco";
            worksheet.Range["A2,C3,D1"].Value = MisspelledWord;

            // Find all misspelled words and mark them with red background
            IRange searchRange = worksheet.Range["A1:D5"];
            IRange misspelledCell = null;
            do
            {
                misspelledCell = searchRange.Find(MisspelledWord, misspelledCell);
                if (misspelledCell == null)
                {
                    break;
                }
                misspelledCell.Interior.Color = System.Drawing.Color.Red;
            } while (true);
        }
    }

}
